### PR TITLE
feat: correct .github workflows from `new` command to work with docker case

### PR DESCRIPTION
**Pull request by dj8yfo**

### CODE REVIEW

File: cargo-near/src/commands/new/new-project-template/.github/workflows/undeploy-staging.yml
Comment (dj8yfo) [inline cargo-near/src/commands/new/new-project-template/.github/workflows/undeploy-staging.yml:5]: according to following documentation https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=closed#pull_request

image

Comment (dj8yfo) [reply cargo-near/src/commands/new/new-project-template/.github/workflows/undeploy-staging.yml:5]: checked that `undeploy-staging.yml` runs on both prs merged/closed

**Comment by dj8yfo** [comment]
[staging ci fails](https://github.com/dj8yfo/workflows_test/actions/runs/10221175487/job/28283101723) from workflow https://github.com/dj8yfo/workflows_test/blob/test_deploy_failing_git_pushed_check_in_pr/.github/workflows/deploy-staging.yml#L38-L55

**Comment by dj8yfo** [comment]
diff in the workflows generated with `main`
```diff
diff --color -r old/.github/workflows/deploy-production.yml new/.github/workflows/deploy-production.yml
18c18
<         run: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/near/cargo-near/releases/download/cargo-near-v0.6.2/cargo-near-installer.sh | sh
---
>         run: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/near/cargo-near/releases/download/cargo-near-v0.6.4/cargo-near-installer.sh | sh
diff --color -r old/.github/workflows/deploy-staging.yml new/.github/workflows/deploy-staging.yml
37c37
<         run: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/near/cargo-near/releases/download/cargo-near-v0.6.2/cargo-near-installer.sh | sh
---
>         run: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/near/cargo-near/releases/download/cargo-near-v0.6.4/cargo-near-installer.sh | sh
38a39,42
>         env:
>           # WASM reproducibility check akin to SourceScan won't be available for staging contracts, deployed from PRs, 
>           # as pull request git refs `refs/pull/NUMBER/merge` are somewhat harder to access and live only as long as PRs do
>           CARGO_NEAR_REPRODUCIBLE_DISABLE_PUSHED_TO_REMOTE_GIT_CHECK: true
diff --color -r old/.github/workflows/undeploy-staging.yml new/.github/workflows/undeploy-staging.yml
3a4
>     # TODO: GitReference::RevPlusRef
```

**Comment by dj8yfo** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
@dj8yfo Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Do you want to apply for monthly streak? Get 8+ score for a single PR this month and receive boost for race-of-sloths!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/dj8yfo)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/dj8yfo)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: cargo-near/src/commands/new/new-project-template/.github/workflows/deploy-staging.yml
Comment (frol) [inline cargo-near/src/commands/new/new-project-template/.github/workflows/deploy-staging.yml:42]: I could not remember why I did not review this PR, but then I remembered that I did and raised the concern offline, so here I am again.

This env variable looks like a hack to me. I'd rather consider having a flag `--allow-dirty` that will use the local folder instead of cloning it from remote.

What do you think?

Comment (dj8yfo) [reply cargo-near/src/commands/new/new-project-template/.github/workflows/deploy-staging.yml:42]: `--skip-git-remote-check` then?, because the repo is clean at the moment of build and dirty isn't the problem

Comment (frol) [reply cargo-near/src/commands/new/new-project-template/.github/workflows/deploy-staging.yml:42]: I mean, I would love to have a command that would build from my project folder (not cloning it, not checking remote), and I think with this command we can kill two birds with one stone

Comment (dj8yfo) [reply cargo-near/src/commands/new/new-project-template/.github/workflows/deploy-staging.yml:42]: that would be longer to implement and more complex code overall, than just `--skip-git-remote-check`.
primary case is still when source code is available for docker build. 
`--skip-git-remote-check` is specifically for this bump with temporary branches in prs for staging builds in workflows, which shouldn't be a frequent case

**Comment by dj8yfo** [comment]
[staging ci fails]() from workflow https://github.com/dj8yfo/workflows_test/blob/<branch_pr_failing>/.github/workflows/deploy-staging.yml#L38-L55

[staging ci passes]() from workflow https://github.com/dj8yfo/workflows_test/blob/<branch_pr_passing>/.github/workflows/deploy-staging.yml#L38-L59

**Comment by dj8yfo** [comment]
output of `diff -r old/.github/workflows new/.github/workflows` (`new` is pr, `old` is latest release)
```diff
diff --color -r old/.github/workflows/deploy-production.yml new/.github/workflows/deploy-production.yml
18c18
<         run: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/near/cargo-near/releases/download/cargo-near-v0.6.2/cargo-near-installer.sh | sh
---
>         run: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/near/cargo-near/releases/download/cargo-near-v0.8.2/cargo-near-installer.sh | sh
diff --color -r old/.github/workflows/deploy-staging.yml new/.github/workflows/deploy-staging.yml
37c37
<         run: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/near/cargo-near/releases/download/cargo-near-v0.6.2/cargo-near-installer.sh | sh
---
>         run: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/near/cargo-near/releases/download/cargo-near-v0.8.2/cargo-near-installer.sh | sh
38a39,42
>         # `--skip-git-remote-check` was used
>         # as pull request git refs `refs/pull/NUMBER/merge` are somewhat harder to access and live only as long as PRs do
>         # 
>         # WASM reproducibility check akin to SourceScan won't be available for staging contracts, deployed from PRs 
40c44
<           cargo near deploy "${{ env.NEAR_CONTRACT_PR_STAGING_ACCOUNT_ID }}" \
---
>           cargo near deploy --skip-git-remote-check "${{ env.NEAR_CONTRACT_PR_STAGING_ACCOUNT_ID }}" \

```

**Review by frol — APPROVED** [review]
@race-of-sloths score 8

**Pull request merged**
